core/src/main/java/hudson/XmlFile.java

```diff
@@ -246,12 +246,12 @@ public boolean exists() {
         return file.exists();
     }
 
-    public void delete() {
-        file.delete();
+    public void delete() throws IOException {
+        Files.deleteIfExists(Util.fileToPath(file));
     }
     
-    public void mkdirs() {
-        file.getParentFile().mkdirs();
+    public void mkdirs() throws IOException {
+        Files.createDirectories(Util.fileToPath(file.getParentFile()));
```

Comment:
Technically an API change, but there are no plugin consumers of this API, and all the consumers in core were already throwing `IOException` anyway.